gradle-plugin/src/proguard/gradle/ProGuardTask.java

```diff
@@ -1384,17 +1379,37 @@ public void proguard()
         loggingManager.captureStandardError(LogLevel.WARN);
 
         // Run ProGuard with the collected configuration.
-        new ProGuard(getConfiguration()).execute();
+        new ProGuard(getConfiguration(getConfigurationFiles().getElements().get())).execute();
+
+    }
 
+    private Provider<Configuration> getConfigurationProvider() {
+        return getConfigurationFiles().getElements().map(configFiles -> {
+            try {
+                return getConfiguration(configFiles);
+            } catch (IOException | ParseException e) {
+                throw new RuntimeException(e);
+            }
+        });
     }
 
+    private synchronized Configuration getConfiguration(Collection<FileSystemLocation> configurationFiles) throws IOException, ParseException {
+        if (configuration.programJars == null) {
+            loadConfiguration(configurationFiles);
+        }
+        return configuration;
+    }
 
     /**
      * Returns the configuration collected so far, resolving files and
      * reading included configurations.
      */
-    private Configuration getConfiguration() throws IOException, ParseException
-    {
+    private Configuration loadConfiguration(Collection<FileSystemLocation> configurationFiles) throws IOException, ParseException {
+        if (configuration.programJars != null) {
+            return configuration;
+        }
+        getLogger().info("Loading ProGuard configuration");
```

Comment:
Maybe delete this log or change to `debug` level?

Reply:
Done in f70c204